From f9c208d4856a6f9ea9b3ae531eeb86ac8ea2bbb4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ji=C5=99=C3=AD=20Klime=C5=A1?= <jklimes@redhat.com>
Date: Tue, 26 Mar 2013 17:56:50 +0100
Subject: [PATCH] editor: set active combo box item in
 ce_page_setup_mac_combo()
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Failing to do so makes gtk_combo_box_get_active() keep returning -1, even if
the combo box entry text was set (until the user manually selects an item).

Signed-off-by: Jiří Klimeš <jklimes@redhat.com>

NOTE:
the upstream patch updated for RHEL6:
ce-page.c -> page-vlan.c
ce_page_setup_mac_combo -> setup_mac_combo()

---
 src/connection-editor/ce-page.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

--- a/network-manager-applet-0.8.1/src/connection-editor/page-vlan.c
+++ b/network-manager-applet-0.8.1/src/connection-editor/page-vlan.c
@@ -304,6 +304,7 @@ setup_mac_combo (CEPage *self, GtkComboB
                  const char *current_mac, char **mac_list)
 {
 	char **iter, *active_mac = NULL;
+	int i, active_idx = -1;
 	int current_mac_len;
 	GtkWidget *entry;
 
@@ -312,7 +313,7 @@ setup_mac_combo (CEPage *self, GtkComboB
 	else
 		current_mac_len = -1;
 
-	for (iter = mac_list; iter && *iter; iter++) {
+	for (iter = mac_list, i = 0; iter && *iter; iter++, i++) {
 #if GTK_CHECK_VERSION (2,24,0)
 		gtk_combo_box_text_append_text (GTK_COMBO_BOX_TEXT (combo), *iter);
 #else
@@ -320,11 +321,16 @@ setup_mac_combo (CEPage *self, GtkComboB
 #endif
 		if (   current_mac
 		    && g_ascii_strncasecmp (*iter, current_mac, current_mac_len) == 0
-		    && ((*iter)[current_mac_len] == '\0' || (*iter)[current_mac_len] == ' '))
+		    && ((*iter)[current_mac_len] == '\0' || (*iter)[current_mac_len] == ' ')) {
 			active_mac = *iter;
+			active_idx = i;
+		}
 	}
 
 	if (current_mac) {
+		/* set active item */
+		gtk_combo_box_set_active (combo, active_idx);
+
 		if (!active_mac) {
 #if GTK_CHECK_VERSION (2,24,0)
 			gtk_combo_box_text_prepend_text (GTK_COMBO_BOX_TEXT (combo), current_mac);
